sok/management/commands/snowball.py

```diff
@@ -155,6 +164,75 @@ def handle_objs(
 						self.echo(abstract)
 				elif choice in {'', 'n', 'no'}:
 					# TODO Import?
+					data = semanticscholar(paper_id)
+					# Add authors to database
+					authors: List[Author] = []
+					first = True
+					cite_key = ''
+					for author in data.get('authors', []):
+						name = author.get('name', '')
+						author, created = Author.objects.get_or_create(name=name)
+						if created:
+							self.echo(f"Added author: {author}")
+						else:
+							self.echo(f"Author '{author}' alreay known")
+						authors.append(author)
+						if first:
+							first = False
+							if name.rindex(' ') > -1:
+								cite_key = name[name.rindex(' '):]
+							else:
+								cite_key = name
```

Comment:
The code seems a bit complex. The variable `first` is not really required, and a check whether `cite_key` is still empty would suffice. Since authors are collected anyway, one could simply extract the last name of the first author afterwards, e. g.:

```suggestion
					for author in data.get('authors', []):
						name = author.get('name', '')
						author, created = Author.objects.get_or_create(name=name)
						if created:
							self.echo(f"Added author: {author}")
						else:
							self.echo(f"Author '{author}' alreay known")
						authors.append(author)
					cite_key = ''
					if authors:
						cite_key = authors[0].name.split(' ')[-1]
```